lib/std/fs/wasi.zig

```diff
@@ -25,13 +25,29 @@ pub const PreopenType = union(PreopenTypeTag) {
     const Self = @This();
 
     pub fn eql(self: Self, other: PreopenType) bool {
-        if (!mem.eql(u8, @tagName(self), @tagName(other))) return false;
+        if (std.meta.activeTag(self) != std.meta.activeTag(other)) return false;
 
         switch (self) {
             PreopenTypeTag.Dir => |this_path| return mem.eql(u8, this_path, other.Dir),
         }
     }
 
+    // Checks whether `other` refers to a subdirectory of `self` and, if so,
+    // returns the relative path to `other` from `self`
+    pub fn get_relative_path(self: Self, other: PreopenType) ?[]const u8 {
```

Comment:
```suggestion
    pub fn getRelativePath(self: Self, other: PreopenType) ?[]const u8 {
```

The rule of thumb for naming is: for calls corresponding 1:1 with syscalls, etc., preserve the name of the source if possible, for any Zig abstraction, camelCase.